mailpoet/lib/AdminPages/AssetsController.php

```diff
@@ -22,101 +23,99 @@ public function __construct(
-      'newsletter_editor',
-      Env::$assetsUrl . '/dist/js/' . $this->renderer->getJsAsset('newsletter_editor.js'),
-      [],
-      Env::$version,
-      true
-    );
+    $this->enqueueJsEntrypoint('newsletter_editor');
+  }
 
-    $this->wp->wpPrintScripts('wp-i18n');
-    $this->wp->wpSetScriptTranslations('newsletter_editor', 'mailpoet');
+  public function setupFormEditorDependencies(): void {
+    $this->enqueueJsEntrypoint('form_editor');
+  }
 
-    /**
-     * The js file needs to be added immediately since the mailpoet_newsletters_editor_initialize hook is dispatched in template files
-     * Update and remove this line in MAILPOET-4930
-     */
-    \wp_scripts()->do_item('newsletter_editor');
+  public function setupSettingsDependencies(): void {
+    $this->enqueueJsEntrypoint('settings');
   }
 
   public function setupAutomationListingDependencies(): void {
-    $this->wp->wpEnqueueScript(
-      'automation',
-      Env::$assetsUrl . '/dist/js/' . $this->renderer->getJsAsset('automation.js'),
-      [],
-      Env::$version,
-      true
-    );
-    $this->wp->wpSetScriptTranslations('automation', 'mailpoet');
+    $this->enqueueJsEntrypoint('automation');
   }
 
   public function setupAutomationTemplatesDependencies(): void {
-    $this->wp->wpEnqueueScript(
-      'automation_templates',
-      Env::$assetsUrl . '/dist/js/' . $this->renderer->getJsAsset('automation_templates.js'),
-      [],
-      Env::$version,
-      true
-    );
-    $this->wp->wpSetScriptTranslations('automation_templates', 'mailpoet');
+    $this->enqueueJsEntrypoint('automation_templates');
   }
 
   public function setupAutomationEditorDependencies(): void {
-    $this->wp->wpEnqueueScript(
-      'automation_editor',
-      Env::$assetsUrl . '/dist/js/' . $this->renderer->getJsAsset('automation_editor.js'),
-      ['wp-date'],
-      Env::$version,
-      true
-    );
-    $this->wp->wpSetScriptTranslations('automation_editor', 'mailpoet');
+    $this->enqueueJsEntrypoint('automation_editor', ['wp-date']);
   }
 
   public function setupAutomationAnalyticsDependencies(): void {
-    $this->wp->wpEnqueueScript(
-      'automation_analytics',
-      Env::$assetsUrl . '/dist/js/' . $this->renderer->getJsAsset('automation_analytics.js'),
-      [],
-      Env::$version,
-      true
-    );
-    $this->wp->wpSetScriptTranslations('automation_analytics', 'mailpoet');
+    $this->enqueueJsEntrypoint('automation_analytics');
 
     $this->wp->wpEnqueueStyle(
       'automation_analytics',
       Env::$assetsUrl . '/dist/css/' . $this->renderer->getCssAsset('mailpoet-automation-analytics.css')
     );
   }
 
-  private function addAdminCommons(): void {
-    $this->wp->wpRegisterScript(
-      'mailpoet_admin_commons',
-      Env::$assetsUrl . '/dist/js/' . $this->renderer->getJsAsset('commons.js'),
-      [],
+  private function enqueueJsEntrypoint(string $asset, array $dependencies = []): void {
+    $name = 'mailpoet_entrypoint';
```

Comment:
I may not be understanding this correctly, but I'm a bit confused about why this is a variable instead of a constant.

Reply:
Fixed by https://github.com/mailpoet/mailpoet/commit/e0e06489231eed0e8b730de0b8ef21b9133940f4. I originally needed a constant name for all entrypoints due to different queuing logic I was playing with. In the current state it doesn't make much of a sense, so I reverted that.